File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.py
class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        nums = []

        rem = grid[0][0] % x

        for row in grid:
            for num in row:
                if num % x != rem:
                    return -1
                nums.append(num)

        nums.sort()

        median = nums[len(nums)//2]

        ops = 0

        for num in nums:
            ops += abs(num-median)//x

        return ops

File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.java
class Solution {
    public int minOperations(int[][] grid, int x) {
        int m = grid.length;
        int n = grid[0].length;
        int size = m * n;

        int[] nums = new int[size];

        int rem = grid[0][0] % x;
        int idx = 0;

        for(int i=0;i<m;i++){
            for(int j=0;j<n;j++){

                int val = grid[i][j];

                if(val % x != rem)
                    return -1;

                nums[idx++] = val;
            }
        }

        Arrays.sort(nums);

        int median = nums[size/2];

        int ops = 0;

        for(int num : nums){
            ops += Math.abs(num - median) / x;
        }

        return ops;
    }
}